packages/demo/src/configuration/blocks/effects/spritesheetBlock.ts

```diff
@@ -0,0 +1,123 @@
+import type { Effect } from "@babylonjs/core/Materials/effect";
+import { type SmartFilter, type IDisableableBlock, type RuntimeData, createStrongRef } from "@babylonjs/smart-filters";
+import { ShaderBlock, ConnectionPointType, ShaderBinding } from "@babylonjs/smart-filters";
+import { BlockNames } from "../blockNames";
+import { shaderProgram, uniforms } from "./spritesheetBlock.shader";
+
+/**
+ * The shader bindings for the Spritesheet block.
+ */
+export class SpritesheetShaderBinding extends ShaderBinding {
+    private readonly _inputTexture: RuntimeData<ConnectionPointType.Texture>;
+    private readonly _time: RuntimeData<ConnectionPointType.Float>;
+    private readonly _rows: RuntimeData<ConnectionPointType.Float>;
+    private readonly _cols: RuntimeData<ConnectionPointType.Float>;
+    private readonly _frames: RuntimeData<ConnectionPointType.Float>;
+
+    /**
+     * Creates a new shader binding instance for the SpriteSheet block.
+     * @param parentBlock - The parent block
+     * @param inputTexture - The input texture
+     * @param time - The time passed since the start of the effect
+     * @param rows - The number of rows in the sprite sheet
+     * @param cols - The number of columns in the sprite sheet
+     * @param frames - The number of frames to show
+     */
+    constructor(
+        parentBlock: IDisableableBlock,
+        inputTexture: RuntimeData<ConnectionPointType.Texture>,
+        time: RuntimeData<ConnectionPointType.Float>,
+        rows: RuntimeData<ConnectionPointType.Float>,
+        cols: RuntimeData<ConnectionPointType.Float>,
+        frames: RuntimeData<ConnectionPointType.Float>
+    ) {
+        super(parentBlock);
+        this._inputTexture = inputTexture;
+        this._time = time;
+        this._rows = rows;
+        this._cols = cols;
+        this._frames = frames;
+    }
+
+    /**
+     * Binds all the required data to the shader when rendering.
+     * @param effect - defines the effect to bind the data to
+     */
+    public override bind(effect: Effect): void {
+        super.bind(effect);
+        effect.setTexture(this.getRemappedName(uniforms.input), this._inputTexture.value);
+        effect.setFloat(this.getRemappedName(uniforms.time), this._time.value);
+        effect.setFloat(this.getRemappedName(uniforms.rows), this._rows.value);
+        effect.setFloat(this.getRemappedName(uniforms.cols), this._cols.value);
+
+        // Apply default value for frame count if it was not provided
+        effect.setFloat(
```

Comment:
Thought it'd just be nice to have the frames connection point be optional and calculate a value for you based on rows/columns